src/mbgl/style/conversion/filter.cpp

```diff
@@ -236,10 +236,13 @@ optional<Filter> convertCompoundFilter(const Convertible& value, Error& error) {
 }
     
 optional<Filter> convertExpressionFilter(const Convertible& value, Error& error) {
-    optional<std::unique_ptr<Expression>> expression = convert<std::unique_ptr<Expression>>(value, error, expression::type::Boolean);
+    expression::ParsingContext ctx(optional<expression::type::Type>{expression::type::Boolean});
```

Comment:
Why is the signature of this constructor being used here `ParsingContext(optional<type::Type> expected_)`? Given that there's also a default constructor, couldn't it simply be `ParsingContext(type::Type expected_)`?

Reply:
Good point - this is probably vestigial; I'll add this change here.